### PR TITLE
Cleanup cases in RankTwoScalarTools

### PR DESCRIPTION
**Pull request by tophmatthews**
(#13927)

### CODE REVIEW

File: modules/tensor_mechanics/include/utils/RankTwoScalarTools.h [hunk 1]
```diff
@@ -424,69 +424,47 @@ getQuantity(const RankTwoTensorTempl<T> & tensor,
             const Point & curr_point,
             Point & direction)
 {
-  T val = 0.0;
-
   switch (scalar_type)
   {
     case 0:
```
Comment (tophmatthews) [inline modules/tensor_mechanics/include/utils/RankTwoScalarTools.h]: ping @dschwen. I was trying to figure out a way to get rid of the ints, but there wasn't a clean way without doing string evaluations (which should be avoided correct?)

Comment (dschwen) [reply modules/tensor_mechanics/include/utils/RankTwoScalarTools.h]: could use enums...

Comment (tophmatthews) [reply modules/tensor_mechanics/include/utils/RankTwoScalarTools.h]: I would have to make a separate enum in the header that matches what is in the .C correct? Not sure if that is more "fool proof" than just making sure the ints match here

Comment (tophmatthews) [reply modules/tensor_mechanics/include/utils/RankTwoScalarTools.h]: Yeah, this is really the only place where there is comparisons to the enum (e.g. vonmises, volumetric). It's probably just fine being ints. Doing the return does clean it up nice though, so should be good to go if this was what you were suggesting earlier?

**Comment by tophmatthews** [comment]
FYI good to go @dschwen, small fixes only

**Review by dschwen — CHANGES_REQUESTED** [review]
Sorry I didn't catch this earlier, but the mooseenum should be passed by reference!

File: modules/tensor_mechanics/include/utils/RankTwoScalarTools.h [hunk 2]
```diff
@@ -418,75 +418,53 @@ stressIntensity(const RankTwoTensorTempl<T> & stress)
 template <typename T>
 T
 getQuantity(const RankTwoTensorTempl<T> & tensor,
-            const MooseEnum scalar_type,
+            const MooseEnum & scalar_type,
             const Point & point1,
             const Point & point2,
             const Point & curr_point,
             Point & direction)
```
Comment (tophmatthews) [inline modules/tensor_mechanics/include/utils/RankTwoScalarTools.h]: It's a bit odd why direction isn't constant. The methods above do change the direction. I hope that is a feature not a bug @dschwen ?

Comment (dschwen) [reply modules/tensor_mechanics/include/utils/RankTwoScalarTools.h]: Yes, direction is an output value

**Review by dschwen — APPROVED** [review]

**Pull request merged**